design/authenticators/authn_oidc/oidc_configuration_enhancements.md

```diff
@@ -0,0 +1,228 @@
+```ruby
+def with_temporary_cert_store(cert_string)
+  Dir.mktmpdir do |temp_dir|
+    # Write the certificate to a temporary file
+    temp_file = File.join(temp_dir, 'ca.pem')
+    File.write(temp_file, cert_string)
+
+    # Compute the X.509 hash of the certificate subject and create a symlink to the temp file
+    cert = OpenSSL::X509::Certificate.new(cert_string)
+    symlink_name = File.join(OpenSSL::X509::DEFAULT_CERT_DIR, "#{cert.subject.hash.to_s(16)}.0")
+    File.symlink(temp_file, symlink_name)
+
+    yield
+  ensure
+    File.unlink(symlink_name) if File.symlink?(symlink_name)
+  end
+end
+```
+
+And its usage:
+```ruby
+::Conjur::CertUtils.with_temporary_cert_store(@ca_cert) do
+  @discovered_provider = @open_id_discovery_service.discover!(@provider_uri)
+end
+```
+
+As for the `HTTPS_PROXY` issue, it appears to be fixed in a more recent version of OpenIDConnect gem which 
+migrated from the `httpclient` library to `faraday` for handling outbound connections. Therefore we should 
+simply be able to bump this gem version (assuming no new issues arise) in order to address this issue.
+
+The last major concern was around better debug logging. This can also be enabled from the Conjur side, 
+via a code snippet like below:
+```
+OpenIDConnect.logger = WebFinger.logger = SWD.logger = Rack::OAuth2.logger = Rails.logger
+OpenIDConnect.debug!
+# Include cert-related debug messages?
+OpenSSL.debug = true
+```
+Running the above in an initializer and will enable debug logging on the OpenIDConnect gem and all its 
+relevant dependencies and ensure that they get routed to the Rails logger. We could also include the debug 
+logs for OpenSSL if we need to specifically focus on certificate-related debugging, which seems to be a 
+common sticking point for authn-oidc users.
+
+### Components
+1. OpenIDConnect - primary gem for [generating an OIDC client](https://github.com/cyberark/conjur/tree/master/app/domain/authentication/authn_oidc/v2/client.rb#L106C20-L106C20) in 
+authn-oidc which can retrieve authorization codes and tokens from the OIDC provider. It outsources discovery 
+to a couple other small gems which we need to consider as well:
+1. SWD (Simple Web Discovery) - handles OIDC configuration discovery against the provider URI configuration 
+endpoint: `/.well-known/openid-configuration` 
+1. WebFinger - used for dynamic discovery when the credential issuer is unknown to the client. **Not used in Conjur.**
+
+### Considerations around OIDC Discovery
+The SWD gem discussed previously is used by OpenIDConnect across many authenticators in Conjur (authn-oidc, 
+authn-jwt, authn-azure, authn-gcp). It performs a couple common tasks in Conjur by invoking the OIDC discovery 
+endpoint:
+  - validate authenticator status and provider connectivity
+  - fetch provider keys for decoding tokens
+
+This should not interfere with the proposed solution, but it is important to note since the modified code will be 
```

Comment:
You go from discussing the SWD gem which was only just referenced above to talking about modified code. Are you intending to modify the gem's code? More is needed about how this modified code could affect the other authns.

Reply:
No gem code would be modified. This is in reference to [this generic OAuth code](https://github.com/cyberark/conjur/tree/master/app/domain/authentication/authn_oidc/v2/client.rb#L106C20-L106C20) which is used by the status webservice of multiple authenticators to verify connectivity to the configured provider. It would need to include the custom cert in the case of authn-oidc without impacting the other authenticators that also use it.